src/main/java/com/educare/unitylend/controller/BorrowRequestController.java

```diff
@@ -44,10 +73,19 @@ ResponseEntity<Boolean> createBorrowRequest(@RequestBody BorrowRequest borrowReq
      */
     @GetMapping("/get-borrow-request-by-user-id/{userId}")
     ResponseEntity<List<BorrowRequest>> getBorrowRequestForUserId(@PathVariable String userId) throws ControllerException {
-        return null;
+        try {
+            if (userId == null || userId.isEmpty()) {
+                log.error("User id cannot be null");
+                return ResponseEntity.badRequest().body(null);
+            }
+            List<BorrowRequest> borrowRequestList = borrowRequestService.getBorrowRequestForUserId(userId);
+            return ResponseEntity.ok(borrowRequestList);
```

Comment:
Add a validation here if `borrowRequestList `is empty, return 200 response code with a message.